### ISSUE TITLE
It appears that there may be a bug in the libvmaf CUDA implementation.

### ISSUE THREAD

**Issue by nukim1** [Netflix/vmaf#1305]
I am currently implementing a program that utilizes multiple instances of libvmaf asynchronously. 

Initializing each instance by setting VmafCudaConfiguration::cu_ctx = cuCtxCreate(...) poses no issues. 

However, when all instances are initialized with the same CUDA context, problems arise during the "vmaf_score_at_index" calls, resulting 
in incorrect VMAF scores or NaN values being output intermittently.

I've checked my code from various angles just in case there might be a problem, but everything seems fine. It is likely that the issue lies with libvmaf.

Moreover, if each libvmaf instance does not share the same CUDA context, there is a performance degradation issue.

this is my one of test result: [](https://www.diffchecker.com/NWoXN3Aa/ )

**Comment by nilfm99** [comment]
I believe this is the same or similar issue to #1180. @kylophone FYI, in case something needs to be fixed

**Comment by nukim1** [comment]
> I believe this is the same or similar issue to #1180. @kylophone FYI, in case something needs to be fixed
I believe my situation differs from what you've mentioned. Although I'm not explicitly mentioned, I called 'vmaf_score_at_index' with a exact two pictures delay for every cases.
Therefore, no log messages were output from any vmaf instancese. Loglevel has been set to DEBUG

**Comment by kylophone** [comment]
Yes, I believe this is a similar synchronization to #1180. I think the only way around this is to introduce a new async callback-type api.

**Comment by lawrencecurtis** [comment]
Has anybody gotten any further with this? 

I believe from my testing this also stops us from running multi libvmaf_cuda filters and renders older Jetson units almost completely useless
